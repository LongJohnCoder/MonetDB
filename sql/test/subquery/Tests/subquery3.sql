CREATE TABLE tbl_ProductSales (ColID int, Product_Category  varchar(64), Product_Name  varchar(64), TotalSales int); 
INSERT INTO tbl_ProductSales VALUES (1,'Game','Mobo Game',200),(2,'Game','PKO Game',400),(3,'Fashion','Shirt',500),(4,'Fashion','Shorts',100);
CREATE TABLE another_T (col1 INT, col2 INT, col3 INT, col4 INT, col5 INT, col6 INT, col7 INT, col8 INT);
INSERT INTO another_T VALUES (1,2,3,4,5,6,7,8), (11,22,33,44,55,66,77,88), (111,222,333,444,555,666,777,888), (1111,2222,3333,4444,5555,6666,7777,8888);

/* Wrong output */
SELECT
    NOT MAX(t1.col6) IN (SELECT SUM(t1.col6) FROM tbl_ProductSales tp HAVING MAX(t1.col1) > MIN(tp.colID))
FROM another_T t1
GROUP BY t1.col6, t1.col7;
	-- True
	-- False
	-- False
	-- False

SELECT
    (SELECT MAX(ColID + col2) FROM tbl_ProductSales) * DENSE_RANK() OVER (PARTITION BY AVG(DISTINCT col5)),
    AVG(col1) * MIN(col8) OVER (PARTITION BY col5 ORDER BY col1 ROWS UNBOUNDED PRECEDING) evil,
    MAX(col3) / 10 + SUM(col1) * 10
FROM another_T
GROUP BY col1, col2, col5, col8;
	-- 6    8       10
	-- 26   968     113
	-- 226  98568   1143
	-- 2226 9874568 11443

SELECT
    CAST (NOT col1 IN (SELECT col2 FROM another_T GROUP BY col2) AS INTEGER) | CAST (col2 IN (SELECT col2 FROM another_T GROUP BY col2) AS INTEGER),
    CAST (EXISTS (SELECT MAX(col5) * MAX(col4) FROM another_T GROUP BY col5, col4) AS INTEGER) & CAST (SUM(col4) IN (SELECT DISTINCT col2 FROM another_T GROUP BY col2) AS INTEGER)
FROM another_T
GROUP BY col1, col2, col5;
	-- 1	0
	-- 1	0
	-- 1	0
	-- 1	0

SELECT
	EXISTS (SELECT col1 WHERE TRUE),
	EXISTS (SELECT col1 WHERE FALSE),
	EXISTS (SELECT col1 WHERE NULL),
	NOT EXISTS (SELECT col1 WHERE TRUE),
	NOT EXISTS (SELECT col1 WHERE FALSE),
	NOT EXISTS (SELECT col1 WHERE NULL)
FROM another_T t1;
	-- True False False False True True
	-- True False False False True True
	-- True False False False True True
	-- True False False False True True

SELECT
	EXISTS (SELECT AVG(col1) WHERE TRUE),
	EXISTS (SELECT AVG(col1) WHERE FALSE),
	EXISTS (SELECT AVG(col1) WHERE NULL),
	NOT EXISTS (SELECT AVG(col1) WHERE TRUE),
	NOT EXISTS (SELECT AVG(col1) WHERE FALSE),
	NOT EXISTS (SELECT AVG(col1) WHERE NULL)
FROM another_T t1;
	-- True False False False True True

SELECT
	EXISTS (SELECT RANK() OVER (PARTITION BY SUM(DISTINCT col5)))
FROM another_T t1;
	-- True

/* BROKEN 
SELECT
    (SELECT AVG(col1) OVER (PARTITION BY col5 ORDER BY col1 ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
FROM another_T t1; --error, more than one row returned by a subquery used as an expression
*/

SELECT
    (SELECT SUM(col2) OVER (PARTITION BY SUM(col2) ORDER BY MAX(col1 + ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
FROM another_T t1
GROUP BY col1; --error, subquery uses ungrouped column "t1.col2" from outer query

/* BROKEN
SELECT
    (SELECT SUM(SUM(col2)) OVER (PARTITION BY SUM(col2) ORDER BY MAX(col2) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
FROM another_T t1
GROUP BY col1; --error, more than one row returned by a subquery used as an expression
*/

SELECT
    (SELECT DENSE_RANK() OVER (PARTITION BY col5 ORDER BY col1) FROM tbl_ProductSales)
FROM another_T t1; --error, more than one row returned by a subquery used as an expression

SELECT
    (SELECT DENSE_RANK() OVER (PARTITION BY MIN(col5) ORDER BY MAX(col8)) FROM tbl_ProductSales)
FROM another_T t1
GROUP BY col6; --error, more than one row returned by a subquery used as an expression

SELECT
    (SELECT DENSE_RANK() OVER (PARTITION BY MIN(col5) ORDER BY col8 * ColID) FROM tbl_ProductSales)
FROM another_T t1
GROUP BY col6; --error, subquery uses ungrouped column "t1.col8" from outer query

/* BROKEN
SELECT
    (SELECT t2.col1 * SUM(SUM(t1.col2)) OVER (PARTITION BY SUM(t1.col2) ORDER BY MAX(t1.col1) ROWS UNBOUNDED PRECEDING) FROM another_T t2)
FROM another_T t1
GROUP BY col1; --error, more than one row returned by a subquery used as an expression
*/

SELECT
    (SELECT t2.col1 * SUM(SUM(col2)) OVER (PARTITION BY SUM(col2) ORDER BY MAX(col1) ROWS UNBOUNDED PRECEDING) FROM another_T t2)
FROM another_T t1
GROUP BY col1; --error, column "t2.col1" must appear in the GROUP BY clause or be used in an aggregate function

SELECT
    (SELECT SUM(AVG(ColID)) OVER (PARTITION BY SUM(ColID) ORDER BY MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales GROUP BY t1.col1)
FROM another_T t1;
	-- 2,5000
	-- 2,5000
	-- 2,5000
	-- 2,5000

SELECT
    (SELECT SUM(AVG(ColID + col1)) OVER (PARTITION BY SUM(ColID + col3) ORDER BY MAX(ColID) * col4 ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales GROUP BY t1.col1)
FROM another_T t1;
	-- 3,5000
	-- 13,5000
	-- 113,5000
	-- 1113,5000

SELECT
    (SELECT MAX(t1.col2) * SUM(AVG(ColID)) OVER (PARTITION BY SUM(ColID) ORDER BY MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
FROM another_T t1;
	-- 5555

SELECT
    (SELECT SUM(AVG(ColID)) OVER (PARTITION BY MAX(t1.col2) * SUM(ColID) ORDER BY MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
FROM another_T t1;
	-- 2,5000

SELECT
    (SELECT SUM(AVG(ColID)) OVER (PARTITION BY SUM(ColID) ORDER BY MAX(t1.col2) * MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
FROM another_T t1;
	-- 2,5000

SELECT
    (SELECT MAX(ColID) * SUM(AVG(ColID)) OVER (PARTITION BY SUM(ColID) ORDER BY MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
FROM another_T t1; --MonetDB outputs this one right, but we should leave it here, as it doesn't trigger an error
	-- 10
	-- 10
	-- 10
	-- 10

SELECT
    (SELECT SUM(SUM(col2)) OVER (PARTITION BY SUM(col2) ORDER BY MAX(col1) ROWS UNBOUNDED PRECEDING) FROM another_T)
FROM another_T t1
GROUP BY col1; --MonetDB outputs this one right, but we should leave it here, as it doesn't trigger an error
	-- 2468
	-- 2468
	-- 2468
	-- 2468

/* We shouldn't allow the following internal functions/procedures to be called from regular queries */
--SELECT "identity"(col1) FROM another_T;
--SELECT "rowid"(col1) FROM another_T;
--SELECT "in"(true, true) FROM another_T;
--SELECT "rotate_xor_hash"(1, 1, 1) FROM another_T;
--CALL sys_update_schemas();
--CALL sys_update_tables();

DROP TABLE tbl_ProductSales;
DROP TABLE another_T;
