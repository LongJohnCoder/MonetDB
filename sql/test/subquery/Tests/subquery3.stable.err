stderr of test 'subquery3` in directory 'sql/test/subquery` itself:


# 17:40:41 >  
# 17:40:41 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33945" "--set" "mapi_usock=/var/tmp/mtest-14741/.s.monetdb.33945" "--forcemito" "--dbpath=/home/niels/scratch/monetdb/Linux-x86_64/var/MonetDB/mTests_sql_test_subquery" "--set" "embedded_c=true"
# 17:40:41 >  

# builtin opt 	gdk_dbpath = /home/niels/scratch/monetdb/Linux-x86_64/var/monetdb5/dbfarm/demo
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_ipv6 = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 33945
# cmdline opt 	mapi_usock = /var/tmp/mtest-14741/.s.monetdb.33945
# cmdline opt 	gdk_dbpath = /home/niels/scratch/monetdb/Linux-x86_64/var/MonetDB/mTests_sql_test_subquery
# cmdline opt 	embedded_c = true
#client2:!ERROR:SQLException:assert:M0M29!zero_or_one: cardinality violation, scalar expression expected
#client2:!ERROR:ParseException:SQLparser:42000!SELECT: cannot use non GROUP BY column 'col2' in query results without an aggregate function
#client2:!ERROR:SQLException:assert:M0M29!zero_or_one: cardinality violation, scalar expression expected
#client2:!ERROR:SQLException:assert:M0M29!zero_or_one: cardinality violation, scalar expression expected
#client2:!ERROR:SQLException:assert:M0M29!zero_or_one: cardinality violation, scalar expression expected
#client2:!ERROR:ParseException:SQLparser:42000!SELECT: cannot use non GROUP BY column 'col8' in query results without an aggregate function
#client2:!ERROR:SQLException:assert:M0M29!zero_or_one: cardinality violation, scalar expression expected
#client2:!ERROR:ParseException:SQLparser:42000!SELECT: cannot use non GROUP BY column in query results without an aggregate function

# 17:40:41 >  
# 17:40:41 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-14741" "--port=33945"
# 17:40:41 >  

MAPI  = (monetdb) /var/tmp/mtest-14741/.s.monetdb.33945
QUERY = SELECT
            (SELECT AVG(col1) OVER (PARTITION BY col5 ORDER BY col1 ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
        FROM another_T t1; --error, more than one row returned by a subquery used as an expression
ERROR = !zero_or_one: cardinality violation, scalar expression expected
CODE  = M0M29
MAPI  = (monetdb) /var/tmp/mtest-14741/.s.monetdb.33945
QUERY = SELECT
            (SELECT SUM(col2) OVER (PARTITION BY SUM(col2) ORDER BY MAX(col1 + ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
        FROM another_T t1
        GROUP BY col1; --error, subquery uses ungrouped column "t1.col2" from outer query
ERROR = !SELECT: cannot use non GROUP BY column 'col2' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-14741/.s.monetdb.33945
QUERY = SELECT
            (SELECT SUM(SUM(col2)) OVER (PARTITION BY SUM(col2) ORDER BY MAX(col2) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
        FROM another_T t1
        GROUP BY col1; --error, more than one row returned by a subquery used as an expression
ERROR = !zero_or_one: cardinality violation, scalar expression expected
CODE  = M0M29
MAPI  = (monetdb) /var/tmp/mtest-14741/.s.monetdb.33945
QUERY = SELECT
            (SELECT DENSE_RANK() OVER (PARTITION BY col5 ORDER BY col1) FROM tbl_ProductSales)
        FROM another_T t1; --error, more than one row returned by a subquery used as an expression
ERROR = !zero_or_one: cardinality violation, scalar expression expected
CODE  = M0M29
MAPI  = (monetdb) /var/tmp/mtest-14741/.s.monetdb.33945
QUERY = SELECT
            (SELECT DENSE_RANK() OVER (PARTITION BY MIN(col5) ORDER BY MAX(col8)) FROM tbl_ProductSales)
        FROM another_T t1
        GROUP BY col6; --error, more than one row returned by a subquery used as an expression
ERROR = !zero_or_one: cardinality violation, scalar expression expected
CODE  = M0M29
MAPI  = (monetdb) /var/tmp/mtest-14741/.s.monetdb.33945
QUERY = SELECT
            (SELECT DENSE_RANK() OVER (PARTITION BY MIN(col5) ORDER BY col8 * ColID) FROM tbl_ProductSales)
        FROM another_T t1
        GROUP BY col6; --error, subquery uses ungrouped column "t1.col8" from outer query
ERROR = !SELECT: cannot use non GROUP BY column 'col8' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-14741/.s.monetdb.33945
QUERY = SELECT
            (SELECT t2.col1 * SUM(SUM(t1.col2)) OVER (PARTITION BY SUM(t1.col2) ORDER BY MAX(t1.col1) ROWS UNBOUNDED PRECEDING) FROM another_T t2)
        FROM another_T t1
        GROUP BY col1; --error, more than one row returned by a subquery used as an expression
ERROR = !zero_or_one: cardinality violation, scalar expression expected
CODE  = M0M29
MAPI  = (monetdb) /var/tmp/mtest-14741/.s.monetdb.33945
QUERY = SELECT
            (SELECT t2.col1 * SUM(SUM(col2)) OVER (PARTITION BY SUM(col2) ORDER BY MAX(col1) ROWS UNBOUNDED PRECEDING) FROM another_T t2)
        FROM another_T t1
        GROUP BY col1; --error, column "t2.col1" must appear in the GROUP BY clause or be used in an aggregate function
ERROR = !SELECT: cannot use non GROUP BY column in query results without an aggregate function
CODE  = 42000

# 17:40:41 >  
# 17:40:41 >  "Done."
# 17:40:41 >  

