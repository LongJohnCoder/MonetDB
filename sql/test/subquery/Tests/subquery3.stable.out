stdout of test 'subquery3` in directory 'sql/test/subquery` itself:


# 17:40:41 >  
# 17:40:41 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33945" "--set" "mapi_usock=/var/tmp/mtest-14741/.s.monetdb.33945" "--forcemito" "--dbpath=/home/niels/scratch/monetdb/Linux-x86_64/var/MonetDB/mTests_sql_test_subquery" "--set" "embedded_c=true"
# 17:40:41 >  

# MonetDB 5 server v11.36.0 (hg id: 1d18bd8860b4+)
# This is an unreleased version
# Serving database 'mTests_sql_test_subquery', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 15.384 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2019 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://localhost.localdomain:33945/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-14741/.s.monetdb.33945
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

# 17:40:41 >  
# 17:40:41 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-14741" "--port=33945"
# 17:40:41 >  

#CREATE TABLE tbl_ProductSales (ColID int, Product_Category  varchar(64), Product_Name  varchar(64), TotalSales int); 
#INSERT INTO tbl_ProductSales VALUES (1,'Game','Mobo Game',200),(2,'Game','PKO Game',400),(3,'Fashion','Shirt',500),(4,'Fashion','Shorts',100);
[ 4	]
#CREATE TABLE another_T (col1 INT, col2 INT, col3 INT, col4 INT, col5 INT, col6 INT, col7 INT, col8 INT);
#INSERT INTO another_T VALUES (1,2,3,4,5,6,7,8), (11,22,33,44,55,66,77,88), (111,222,333,444,555,666,777,888), (1111,2222,3333,4444,5555,6666,7777,8888);
[ 4	]
#SELECT
#    NOT MAX(t1.col6) IN (SELECT SUM(t1.col6) FROM tbl_ProductSales tp HAVING MAX(t1.col1) > MIN(tp.colID))
#FROM another_T t1
#GROUP BY t1.col6, t1.col7;
% .%30 # table_name
% %30 # name
% boolean # type
% 5 # length
[ false	]
[ false	]
[ false	]
[ true	]
#SELECT
#    (SELECT MAX(ColID + col2) FROM tbl_ProductSales) * DENSE_RANK() OVER (PARTITION BY AVG(DISTINCT col5)),
#    AVG(col1) * MIN(col8) OVER (PARTITION BY col5 ORDER BY col1 ROWS UNBOUNDED PRECEDING) evil,
#    MAX(col3) / 10 + SUM(col1) * 10
#FROM another_T
#GROUP BY col1, col2, col5, col8;
% .%4,	.%37,	.%10 # table_name
% %4,	evil,	%10 # name
% bigint,	double,	bigint # type
% 4,	24,	5 # length
[ 6,	8,	10	]
[ 26,	968,	113	]
[ 226,	98568,	1143	]
[ 2226,	9874568,	11443	]
#SELECT
#    CAST (NOT col1 IN (SELECT col2 FROM another_T GROUP BY col2) AS INTEGER) | CAST (col2 IN (SELECT col2 FROM another_T GROUP BY col2) AS INTEGER),
#    CAST (EXISTS (SELECT MAX(col5) * MAX(col4) FROM another_T GROUP BY col5, col4) AS INTEGER) & CAST (SUM(col4) IN (SELECT DISTINCT col2 FROM another_T GROUP BY col2) AS INTEGER)
#FROM another_T
#GROUP BY col1, col2, col5;
% .%51,	.%52 # table_name
% %51,	%52 # name
% int,	int # type
% 1,	1 # length
[ 1,	0	]
[ 1,	0	]
[ 1,	0	]
[ 1,	0	]
#SELECT
#	EXISTS (SELECT col1 WHERE TRUE),
#	EXISTS (SELECT col1 WHERE FALSE),
#	EXISTS (SELECT col1 WHERE NULL),
#	NOT EXISTS (SELECT col1 WHERE TRUE),
#	NOT EXISTS (SELECT col1 WHERE FALSE),
#	NOT EXISTS (SELECT col1 WHERE NULL)
#FROM another_T t1;
% .%240,	.%241,	.%242,	.%243,	.%244,	.%245 # table_name
% %240,	%241,	%242,	%243,	%244,	%245 # name
% boolean,	boolean,	boolean,	boolean,	boolean,	boolean # type
% 5,	5,	5,	5,	5,	5 # length
[ true,	false,	false,	false,	true,	true	]
[ true,	false,	false,	false,	true,	true	]
[ true,	false,	false,	false,	true,	true	]
[ true,	false,	false,	false,	true,	true	]
#SELECT
#	EXISTS (SELECT AVG(col1) WHERE TRUE),
#	EXISTS (SELECT AVG(col1) WHERE FALSE),
#	EXISTS (SELECT AVG(col1) WHERE NULL),
#	NOT EXISTS (SELECT AVG(col1) WHERE TRUE),
#	NOT EXISTS (SELECT AVG(col1) WHERE FALSE),
#	NOT EXISTS (SELECT AVG(col1) WHERE NULL)
#FROM another_T t1;
% .%240,	.%241,	.%242,	.%243,	.%244,	.%245 # table_name
% %240,	%241,	%242,	%243,	%244,	%245 # name
% boolean,	boolean,	boolean,	boolean,	boolean,	boolean # type
% 5,	5,	5,	5,	5,	5 # length
[ true,	false,	false,	false,	true,	true	]
#SELECT
#	EXISTS (SELECT RANK() OVER (PARTITION BY SUM(DISTINCT col5)))
#FROM another_T t1;
% .%24 # table_name
% %24 # name
% boolean # type
% 5 # length
[ true	]
#SELECT
#    (SELECT SUM(AVG(ColID)) OVER (PARTITION BY SUM(ColID) ORDER BY MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales GROUP BY t1.col1)
#FROM another_T t1;
% .%5 # table_name
% %5 # name
% double # type
% 24 # length
[ 2.5	]
[ 2.5	]
[ 2.5	]
[ 2.5	]
#SELECT
#    (SELECT SUM(AVG(ColID + col1)) OVER (PARTITION BY SUM(ColID + col3) ORDER BY MAX(ColID) * col4 ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales GROUP BY t1.col1)
#FROM another_T t1;
% .%6 # table_name
% %6 # name
% double # type
% 24 # length
[ 3.5	]
[ 13.5	]
[ 113.5	]
[ 1113.5	]
#SELECT
#    (SELECT MAX(t1.col2) * SUM(AVG(ColID)) OVER (PARTITION BY SUM(ColID) ORDER BY MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
#FROM another_T t1;
% .%5 # table_name
% %5 # name
% double # type
% 24 # length
[ 5555	]
#SELECT
#    (SELECT SUM(AVG(ColID)) OVER (PARTITION BY MAX(t1.col2) * SUM(ColID) ORDER BY MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
#FROM another_T t1;
% .%5 # table_name
% %5 # name
% double # type
% 24 # length
[ 2.5	]
#SELECT
#    (SELECT SUM(AVG(ColID)) OVER (PARTITION BY SUM(ColID) ORDER BY MAX(t1.col2) * MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
#FROM another_T t1;
% .%6 # table_name
% %6 # name
% double # type
% 24 # length
[ 2.5	]
#SELECT
#    (SELECT MAX(ColID) * SUM(AVG(ColID)) OVER (PARTITION BY SUM(ColID) ORDER BY MAX(ColID) ROWS UNBOUNDED PRECEDING) FROM tbl_ProductSales)
#FROM another_T t1; --MonetDB outputs this one right, but we should leave it here, as it doesn't trigger an error
% .%5 # table_name
% %5 # name
% double # type
% 24 # length
[ 10	]
[ 10	]
[ 10	]
[ 10	]
#SELECT
#    (SELECT SUM(SUM(col2)) OVER (PARTITION BY SUM(col2) ORDER BY MAX(col1) ROWS UNBOUNDED PRECEDING) FROM another_T)
#FROM another_T t1
#GROUP BY col1; --MonetDB outputs this one right, but we should leave it here, as it doesn't trigger an error
% .%5 # table_name
% %5 # name
% bigint # type
% 4 # length
[ 2468	]
[ 2468	]
[ 2468	]
[ 2468	]
#DROP TABLE tbl_ProductSales;
#DROP TABLE another_T;

# 17:40:41 >  
# 17:40:41 >  "Done."
# 17:40:41 >  

